packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { FC, ChangeEvent, useState, useEffect } from 'react';
+import { MagnifyingGlass } from '@phosphor-icons/react';
+import useAutocomplete from '@mui/base/useAutocomplete';
+
+import { ComponentType } from '../types';
+import { CWTag } from './CWTag';
+
+import 'components/component_kit/new_designs/CWSearchBar.scss';
+import { IconName } from '../cw_icons/cw_icon_lookup';
+import { getClasses } from '../helpers';
+import { ValidationStatus } from '../cw_validation_text';
+
+type BaseSearchBarProps = {
+  autoComplete?: string;
+  autoFocus?: boolean;
+  containerClassName?: string;
+  defaultValue?: string | number;
+  value?: string | number;
+  iconLeft?: IconName;
+  iconLeftonClick?: () => void;
+  inputValidationFn?: (value: string) => [ValidationStatus, string] | [];
+  label?: string | React.ReactNode;
+  maxLength?: number;
+  name?: string;
+  onInput?: (e) => void;
+  onenterkey?: (e) => void;
+  onClick?: (e) => void;
+  placeholder?: string;
+  tabIndex?: number;
+  manualStatusMessage?: string;
+  manualValidationStatus?: ValidationStatus;
+};
+
+type InputStyleProps = {
+  inputClassName?: string;
+  disabled?: boolean;
+};
+
+type InputInternalStyleProps = {
+  hasLeftIcon?: boolean;
+  isTyping?: boolean;
+};
+
+type SearchBarProps = BaseSearchBarProps &
+  InputStyleProps &
+  InputInternalStyleProps &
+  React.HTMLAttributes<HTMLDivElement>;
+
+const communities = [
+  'Altitude',
+  'Terra Classic',
+  'Osmosis',
+  'Qwoyn Network',
+  '1inch',
+  'Stargate Finance',
+  'Timeless',
+  'Terra Agora',
+  'Injective',
+  'Juno',
+  'Common',
+];
+
+export const CWSearchBar: FC<SearchBarProps> = ({ disabled, placeholder }) => {
```

Comment:
again, how would you like to manage the data of this component from outside of the component? You are missing props that will handle that eg:
- you don't want to have the list of communities defined in this file. You want to fetch it from the API and pass it to the searchbar component by props eg `options`
- if you select some community you would like to react on that outside of the component, probably filter the threads by this community, so you need to expose `selectedItem` outside of this component
- also, this CWSearchBar and CWTag should not be bounded to the community, they should be generic, so we should use `options`, `items`, `selectedItem` itd instead of `communityName`, `community` etc

Reply:
The list of communities (chains) is now passed as props to the component as follows
```typescript
<CWSearchBar
  options={app.config.chains.getAll()}
  placeholder="Search Common"
/>
```

`seletecItem` yet to be implemented.